controllers/configurationpolicy_controller.go

```diff
@@ -335,15 +335,94 @@ func getTemplateConfigErrorMsg(err error) string {
 	return fmt.Sprintf("An unexpected error occurred when configuring the template resolver: %v", err)
 }
 
+type objectTemplateDetails struct {
+	kind         string
+	name         string
+	namespace    string
+	isNamespaced bool
+}
+
+// getObjectTemplateDetails retrieves values from the object templates and returns an array of
+// objects containing the retrieved values.
+// It also gathers namespaces for this policy if necessary:
+//   If a namespaceSelector is present AND objects are namespaced without a namespace specified
+func (r *ConfigurationPolicyReconciler) getObjectTemplateDetails(
+	plc policyv1.ConfigurationPolicy,
+) ([]objectTemplateDetails, []string, bool, error) {
+	templateObjs := make([]objectTemplateDetails, len(plc.Spec.ObjectTemplates))
+	selectedNamespaces := []string{}
+	queryNamespaces := false
+
+	for idx, objectT := range plc.Spec.ObjectTemplates {
+		unstruct, err := unmarshalFromJSON(objectT.ObjectDefinition.Raw)
+		if err != nil {
+			return templateObjs, selectedNamespaces, false, err
+		}
+
+		templateObjs[idx].isNamespaced = r.isObjectNamespaced(&unstruct, true)
+		templateObjs[idx].kind = unstruct.GetKind()
+		templateObjs[idx].name = unstruct.GetName()
+		templateObjs[idx].namespace = unstruct.GetNamespace()
+
+		if templateObjs[idx].isNamespaced && templateObjs[idx].namespace == "" {
+			queryNamespaces = true
+		}
+	}
+
+	// If required, query for namespaces specified in NamespaceSelector for objects to use
+	if queryNamespaces {
+		// Retrieve the namespaces based on filters in NamespaceSelector
+		selector := plc.Spec.NamespaceSelector
+		// If MatchLabels/MatchExpressions/Include were not provided, return no namespaces
+		if selector.MatchLabels == nil && selector.MatchExpressions == nil && len(selector.Include) == 0 {
+			log.Info("NamespaceSelector is empty. Skipping namespace retrieval.")
+		} else {
+			// If an error occurred in the NamespaceSelector, update the policy status and abort
+			var err error
+			selectedNamespaces, err = common.GetSelectedNamespaces(selector)
+			if err != nil {
+				errMsg := "Error filtering namespaces with provided namespaceSelector"
+				log.Error(
+					err, errMsg,
+					"namespaceSelector", fmt.Sprintf("%+v", selector))
+
+				reason := "NamespaceSelector error"
```

Comment:
Optional: This would match the case of the YAML for a user-facing message.
```suggestion
				reason := "namespaceSelector error"
```